src/elements/components/download-button.tsx

```diff
@@ -36,43 +42,147 @@ const getHostFromUrl = (url: string) => {
         }
         return parsedUrl.hostname;
     } catch (e) {
-        console.error(e);
+        console.debug(e);
         return 'an unknown hoster';
     }
 };
 
-export const DownloadButton = ({ url, resource }: DownloadButtonProps) => {
-    const isExternal = !url.startsWith(
-        'https://objectstorage.us-phoenix-1.oraclecloud.com/n/ax6ygfvpvzka/b/open-modeldb-files/'
-    );
-    const host = getHostFromUrl(url);
+const getHostIcon = (host: string) => {
+    switch (host) {
+        case 'GitHub':
+            return <SiGithub className="block" />;
+        case 'Google Drive':
+            return <SiGoogledrive className="block" />;
+        case 'OneDrive':
+            return <SiMicrosoftonedrive className="block" />;
+        case 'Mega':
+            return <SiMega className="block" />;
+        case 'Dropbox':
+            return <SiDropbox className="block" />;
+        default:
+            return <FiExternalLink className="block" />;
+    }
+};
+
+const getIsExternal = (url: string) => {
```

Comment:
Please remove the `get`... It's silly... Just call it `const isMirrorExternal = isExternal(selectedMirror);` or something...